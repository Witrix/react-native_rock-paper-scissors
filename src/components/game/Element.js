import React, { Component } from 'react';
import {
    View,
    Image,
    Text, StyleSheet,
} from 'react-native';
import MyLocalization       from "../../my-lib/MyLocalzation";
import MyRules              from '../../my-lib/MyRules';

export default class Element extends Component {

    constructor(props) {
        super(props);
        this.state = {
            type: props.type,
            mode: props.mode,
        };
    }


    render() {
        let image = null;
        let name = "";
        let element = MyRules.getElement(this.state.type, this.state.mode);
        if (element) {
            image = element.resourceKey;
            name = MyLocalization.getString(element.localizedKey);
        }
        return (
            <View style={style.container}>
                <Image source={image} style={style.image}/>
                <Text>{name}</Text>
            </View>
        );
    }
}

const style = StyleSheet.create({
    container: {
        justifyContent: 'center',
        alignItems: 'center'
    },
    image: {
        width: 70,
        height: 70,
    }
});
