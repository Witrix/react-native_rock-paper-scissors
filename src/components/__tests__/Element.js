import 'react-native';
import React      from 'react';
import Element    from '../game/Element';
import renderer   from 'react-test-renderer';

it('Render correctly: No props', () => {
  const tree = renderer.create(
    <Element />
  );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Rock', () => {
    const tree = renderer.create(
        <Element type="rock"/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Paper', () => {
    const tree = renderer.create(
        <Element type="paper"/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Scissors', () => {
    const tree = renderer.create(
        <Element type="scissors"/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Lizard', () => {
    const tree = renderer.create(
        <Element type="lizard"/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Spock', () => {
    const tree = renderer.create(
        <Element type="spock"/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Invalid props', () => {
    const tree = renderer.create(
        <Element type={-1}/>
    );
    expect(tree).toMatchSnapshot();
});

