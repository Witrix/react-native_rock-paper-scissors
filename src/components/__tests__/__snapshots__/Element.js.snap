// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly: Invalid props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={null}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
</View>
`;

exports[`Render correctly: Lizard 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={null}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
</View>
`;

exports[`Render correctly: No props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={null}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
</View>
`;

exports[`Render correctly: Paper 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Paper
  </Text>
</View>
`;

exports[`Render correctly: Rock 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Rock
  </Text>
</View>
`;

exports[`Render correctly: Scissors 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Scissors
  </Text>
</View>
`;

exports[`Render correctly: Spock 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={null}
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
</View>
`;
